### PR TITLE
add/remove items from userlists

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
#3819 

#### What's this PR do?
This PR:
- Adds 'add to userlist' functionality on resource cards
- renames the old resource card to LearningResourceCardTemplate. This is display only and does not have associated user interactions. Adds a new LearningResourceCard that triggers the drawer and 'add to userlist' dialog.

#### How should this be manually tested?
1. As an anonymous user, visit https://od.odl.local:8063/infinite/search. Resource cards should **not** have a bookmark icon for adding to list.
2. As a logged-in user, visit https://od.odl.local:8063/infinite/search. Resource cards **should** have a bookmark icon for adding to list.
    - click the bookmark icon
    - add/remove a resource to a list
    - also try creating a new list from the 'add to list' dialog.
3. Visit https://od.odl.local:8063/infinite/lists, and check that your items have been added to the list.

### CODE REVIEW

File: frontends/infinite-corridor/src/api/learning-resources/hooks.test.tsx
Comment (ChristopherChudzicki) [inline frontends/infinite-corridor/src/api/learning-resources/hooks.test.tsx:1]: This file adds tests for functionality in this PR, and adds some tests for functionality added in#3835.

**Review by mbertrand — COMMENTED** [review]
One thing I noticed, maybe it is being reserved for a future PR - after I add a resource to a list, the bookmark icon is still white.  From the `/learn` endpoint, bookmarks are black if they are included in a user's lists:

infinite:

Screenshot 2023-03-29 at 8 19 08 AM

learn:

Screenshot 2023-03-29 at 8 19 18 AM

**Review by mbertrand — COMMENTED** [review]
LGTM, just the one question about changing bookmark color for resources in lists.

File: frontends/infinite-corridor/src/components/LearningResourceCard.tsx [hunk 1]
```diff
@@ -40,10 +42,11 @@ const LearningResourceCard: React.FC<LearningResourceCardProps> = ({
         footerActionSlot={
           user.is_authenticated && (
             <IconButton
+              size="small"
               aria-label="Add to list"
               onClick={() => addToList.open(resource)}
             >
-              <BookmarkBorderIcon />
+              {isInList ? <BookmarkIcon /> : <BookmarkBorderIcon />}
```
Comment (mbertrand) [inline frontends/infinite-corridor/src/components/LearningResourceCard.tsx:49]: Under `/learn/`, if I add a resource to a list, the bookmark is filled in right away, but under `/infinite` I have to reload the page or redo a search before the bookmark is filled in.

**Review by mbertrand — CHANGES_REQUESTED** [review]

File: frontends/infinite-corridor/src/components/LearningResourceCard.tsx (hunk 1 above)
Comment (mbertrand) [reply frontends/infinite-corridor/src/components/LearningResourceCard.tsx:49]: fine to fix in a future PR

**Review by mbertrand — APPROVED** [review]
👍

File: frontends/infinite-corridor/src/components/LearningResourceCard.tsx (hunk 1 above)
Comment (ChristopherChudzicki) [reply frontends/infinite-corridor/src/components/LearningResourceCard.tsx:49]: https://github.com/mitodl/open-discussions/issues/3908

**Pull request merged**